contracts/contracts/ValidatorSet.sol

```diff
@@ -29,6 +29,8 @@ contract ValidatorSet {
     mapping(address => AddressStatus) status;
     bool public finalized;  // Was the last validator change finalized. Implies currentValidators == pendingValidators
     address public systemAddress = 0xffffFFFfFFffffffffffffffFfFFFfffFFFfFFfE;
+    uint[] public epochStartHeights;
+    mapping(uint => address[]) public epochStartToValidatorsMapping;
```

Comment:
you could name this `epochValidators` or `validatorsOfEpoch`

Reply:
Okay, if you already suggest I take the first one. I like it when variables which are somehow related have the same prefix. :grin: